### DISCUSSION TITLE
Error: Connection closed in production

### DISCUSSION THREAD

**Discussion by Bilboramix** [vercel/next.js#54186] [Help]
### Summary

My app is randomly crashing and throwing this error. It's totally random...
![image](https://github.com/vercel/next.js/assets/91626615/13fd2303-946a-4102-a8c0-e157861542c0)

Here's the error with `productionBrowserSourceMaps: true`
![image](https://github.com/vercel/next.js/assets/91626615/dfac37ba-d55c-4ae3-a69a-faa5d841b8f7)

Locally i don't have any problem.

The crash is not consistent and i cannot tell if there's something wrong with my code or if it's vercel...

It's the same issue as here : https://github.com/vercel/next.js/issues/52824
and here : https://github.com/vercel/next.js/issues/49205

Some people says it's not fixed, some say it's fixed...
I developped the project on Next 13.4.12 and i upgraded to 13.4.16.
What should i do to at least have a better stacktrace ?

I'm gonna try to reproduce with minimal things, i'll drop the link to the repo here

### Deps
```json
{
  "dependencies": {
    "argon2": "^0.30.3",
    "cloudinary": "^1.40.0",
    "eslint": "8.46.0",
    "eslint-config-next": "13.4.12",
    "mongoose": "^7.4.1",
    "next": "^13.4.16",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "sharp": "^0.32.4",
    "typescript": "5.1.6",
    "uid2": "^1.0.0"
  },
  "devDependencies": {
    "@types/node": "20.4.5",
    "@types/react": "18.2.17",
    "@types/react-dom": "18.2.7",
    "@types/uid2": "^0.0.0",
    "sass": "^1.64.2"
  }
}
```

### Other infos and code snippet
i'm using server actions and my pages follow this pattern :
```ts
export default async function Page() {
  const client = await authenticate();
  if (!client || !client.isAdmin) redirect("/login");

  const webContentBlocks: TWebContentBlockDocument[] = await WebContentBlock.find().sort({ position: 1 });

  return (
    <section>
      <WebContentBlockManager webContentBlocks={webContentBlocks} />
    </section>
  );
}
```

**Comment by verz0** [comment]
hey were you able to find a fix? I have the same problem. It works fine locally but when I deploy it on vercel, 'error-connection closed' occurs

**Reply by Bilboramix** [reply]
Hello, yeah it was actually a crash from my server side. If i remember well it was my database that was not connected on a specific call.
The error i get here is generic and as i said i do not have any clear stack trace of it in production but i guess i was able to reproduce it in local after a while.
Cannot help more here. If you got some promise to await or things like that, you may look into this kind of code.
